source/Calamari/Kubernetes/Commands/Executors/GlobberGrouping.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Calamari.Common.Plumbing.FileSystem;
+using Octopus.CoreUtilities.Extensions;
+
+namespace Calamari.Kubernetes.Commands.Executors
+{
+    public class GlobberGrouping
+    {
+        readonly ICalamariFileSystem fileSystem;
+        const string GroupedDirectoryName = "grouped";
+        public GlobberGrouping(ICalamariFileSystem fileSystem)
+        {
+            this.fileSystem = fileSystem;
+        }
+
+        /// <summary>
+        /// Groups together files from glob patterns into directories.
+        /// </summary>
+        /// <param name="deployment"></param>
+        /// <param name="rootDirectory"></param>
+        /// <param name="globs"></param>
+        /// <returns>Ordered result of directories</returns>
```

Comment:
nit: Just a little clearer about the ordering this spits out

```suggestion
        /// <returns>Result of directories in the same order as globs parameter</returns>
```